xclim/subset.py

```diff
@@ -458,11 +457,10 @@ def subset_shape(
     """
     wgs84 = CRS("+proj=longlat +ellps=WGS84 +datum=WGS84 +no_defs")
 
-    # TODO : edge case using polygon splitting decorator touches original ds when subsetting?
     if isinstance(ds, xarray.DataArray):
-        ds_copy = copy.deepcopy(ds._to_temp_dataset())
+        ds_copy = ds._to_temp_dataset()
```

Comment:
This is enough to prevent overflowing the memory?

Reply:
Deepcopy was copying everything included the data, but  `ds.copy` is only creating a new Dataset with views to the same data. Same for `_to_temp_dataset()` in the case of DataArrays. When we reassign inplace further down, the Dataset replaces the view by a  new one, not touching the input data.

Reply:
Useful to know. This could come in handy in other ways.